### PR TITLE
chore: function overloads implementation of startActiveSpan in noop t…

### PR DESCRIPTION
**Pull request by naseemkullah**
…racer

Signed-off-by: naseemkullah <naseem@transit.app>

### CODE REVIEW

**Review by vmarchaud — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

File: src/trace/NoopTracer.ts [hunk 1]
```diff
@@ -49,41 +49,47 @@ export class NoopTracer implements Tracer {
 
   startActiveSpan<F extends (span: Span) => ReturnType<F>>(
     name: string,
-    arg2: F | SpanOptions,
+    fn: F
+  ): ReturnType<F>;
+  startActiveSpan<F extends (span: Span) => ReturnType<F>>(
+    name: string,
+    opts: SpanOptions,
```
Comment (dyladan) [inline src/trace/NoopTracer.ts:56]:
```suggestion
    opts: SpanOptions | undefined,
```

File: src/trace/NoopTracer.ts [hunk 2]
```diff
@@ -49,41 +49,47 @@ export class NoopTracer implements Tracer {
...
+    fn: F
+  ): ReturnType<F>;
+  startActiveSpan<F extends (span: Span) => ReturnType<F>>(
+    name: string,
+    opts: SpanOptions,
+    ctx: Context,
```
Comment (dyladan) [inline src/trace/NoopTracer.ts:62]:
```suggestion
    opts: SpanOptions | undefined,
    ctx: Context | undefined,
```

**Review by vmarchaud — APPROVED** [review]

**Review by obecny — APPROVED** [review]
lgtm

**Comment by naseemkullah** [comment]
> The base branch restricts merging to authorized users.

Could someone with authority please merge?

**Comment by obecny** [comment]
> > The base branch restricts merging to authorized users.
> 
> Could someone with authority please merge?

on api repo we want all maintainers to approve.

**Review by dyladan — APPROVED** [review]

**Pull request merged**